src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_safe_repr.py

```diff
@@ -304,6 +304,13 @@ def _repr_obj(self, obj, level, limit_inner, limit_outer):
         # you are using the wrong class.
         left_count, right_count = max(1, int(2 * limit / 3)), max(1, int(limit / 3))  # noqa
 
-        yield obj_repr[:left_count]
-        yield '...'
-        yield obj_repr[-right_count:]
+        if isintance(obj_repr, bytes):
+            # convert to unicode before slicing strings
+            obj_repr = unicode(obj_repr, 'utf-8', errors='replace')
+            yield obj_repr[:left_count].encode('utf-8')
+            yield '...'
+            yield obj_repr[-right_count:].encode('utf-8')
```

Comment:
Another note here, we shouldn't really be using `obj_repr.__repr__`, rather, we should use `repr(obj_repr)` and catching errors in case the user-provided `__repr__` is not correct.

Reply:
I think `decode` would use `sys.getdefaultencoding() ` by default.

Reply:
On Python 2, it does if called with no arguments. On Python 3 it defaults to UTF-8, but it should never return `bytes` from `repr()`, so we should be good.

Reply:
Actually I would like to decode from utf-8 instead of default, or at least have try to decode from default and then from utf-8. Because in PyCharm debugger works without `sys.setdefaultencoding()` and I didn't have any issues with `__repr__()` returning utf-8 bytestring so far.

Reply:
Unicode is a mess in Python 2 in general, so there's no single right answer here. The best we can do is try to be consistent with what the language does elsewhere. In this case, there are two possibilities.

First one is to look at the implicit conversion from `unicode` to `str`. The fact that it's there in the code means that somebody is using it. Most likely inadvertently, when repr for an object just concatenates a bunch of strings, and one of them happens to be Unicode. In this case, implicit conversion to `unicode` will use the default encoding, so it makes sense for `repr()` to decode accordingly. To correctly support code that's doing that, we need to do the same.

On the other hand, we can look at what the standard Python REPL does when it prints objects out. And it basically just dumps them to stdout as is, which implies that the expected encoding is whatever `sys.stdout.encoding` is. If you're on Unix, `sys.stdout.encoding` is almost certainly UTF-8. But on Windows, it will be the current console codepage, which is usually not UTF-8. Consequently, `__repr__` that always returns UTF-8 will not print correctly there. And conversely, if we treat it as if it were UTF-8, then any library that encodes it using `sys.stdout.encoding` - specifically so that it prints correctly - will not display correctly in debugger. Looking around, there's some code with e.g. `return (...).encode(sys.stdout.encoding or 'utf-8')` out there.

And since default encoding and stdout encoding are two distinct things, this also means that `__repr__` that returns `unicode` won't always print correctly, either. So the language isn't fully consistent with itself. 

I think of these two, we should go with `sys.stdout.encoding` if present (and default to UTF-8 otherwise). It makes more sense, since the default encoding is almost always ASCII, and changing it is not recommended. 

Would this also solve the problem in your scenario?

Reply:
Side note: Jupyter on Python 2 treats repr of outputs as UTF-8. But it also sets `sys.stdout.encoding` to UTF-8, so it's fully consistent with this behavior.

Reply:
I think using `sys.stdout.encoding or 'utf-8'` will work in my case because I have UTF-8 everywhere.

Reply:
Perfect, looks like we can make everybody happy! :)

One slight problem with that exact idiom - I looked it up, and `sys.stdout` is not guaranteed to have `encoding` at all, if it is replaced (i.e. you might get `AttributeError` exception instead of `None`). So I think it'll need to be:
```py
getattr(sys.stdout, 'encoding', None) or 'utf-8'
```
This will cover both the missing and the present-but-None case.